rules/operation-builder.js

```diff
@@ -0,0 +1,481 @@
+const parse_duration = require('parse-duration');
+const items = require('../items/items');
+
+/**
+ * Operation to execute as part of a rule
+ * @hideconstructor
+ */
+class OperationBuilder {
+    constructor(builder, fn) {
+        this.builder = builder;
+        this.fn = fn;
+    }
+
+    _finishErr() {
+        if (this.fn) {
+            throw new Error("rule already completed")
+        }
+    }
+
+    _then(operation, group, name, description) {
+        this.builder.name = name;
+        this.builder.description = description;
+        this.builder.setOperation(operation, group);
+    }
+
+    /**
+     * Build this rule
+     *
+     * @param {string} name of the rules
+     * @param {string} description of the rule
+     */
+    build(name, description) {
+        if (!this.fn) {
+            throw new Error("Cannot call build without function")
+        }
+        this._then(this.fn, this.group, name, description);
+    }
+
+    /**
+     * Specify the rule group for this rule
+     * 
+     * @param {string} group the group this rule belongs to.
+     * @returns {OperationBuilder} this
+     */
+    inGroup(group) {
```

Comment:
@jpg0 I believe the rule group functionality is something not native to openHAB, and that this library is artificially constructing this feature, correct?  I'm wondering if this should be removed (and the rule toggle features) for simplicity sake?  While a nice feature, it seems like it would be out of scope for what is the official library/API which shouldn't add such new functionality on its own.